Modernize conf.py files to only include modified Copyright, eliminati…

rkent opened the pull request:
Generated-by: Portions of this commit may include code completion from github.copilot version 1.372.0 or later

## Description

Replace or creates a conf.py in each package in this repo, that only generates a custom Copyright notice.

Modern rosdoc2 requires nothing in a user-supplied conf.py, so any additions should be intended to modify standard rosdoc2 behavior. As far as I can tell, the only intentional change to standard rosdoc2 behavior in this repo was to provide a more detailed copyright notice.

Remove cruft such as:
- old stuff that used to be required by rosdoc2 in a custom conf.py
- various configurations of documentation systems (like Latex) that are not used in current ROS 2 documentation infrastructure (See discussion in #833)
- including additional sphinx extensions that, AFAICT, are not actually used by the current geometry2 documentation generation

What remains is an identical conf.py file in each package that generates the custom copyright notice. In some cases this involved modifying an existing conf.py, in others a new conf.py was included.

Package tf2_kdl uses a custom documentation folder name that is not being picked up by rosdoc2. So although this PR includes the new conf.py, until tf2_kdl is modified this is not actually affecting the documentation generation. This will be fixed in a future PR.

Review should confirm that the generated copyright notice is indeed what is desired. The notice is:
`© Copyright 2008-2025, Open Source Robotics Foundation, Inc.`

Hopefully I have not misunderstood the intentions of existing conf.py files, but they appear to me to be mostly automatically-generated boilerplate that really is not needed.

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.*
-->

Partially addresses #833

### Is this user-facing behavior change?

No, only documentation.

### Did you use Generative AI?

See header.

rkent committed:
Modernize conf.py files to only include modified Copyright, eliminating cruft.

Signed-off-by: R Kent James <kent@caspia.com>
Generated-by: Portions of this commit may include code completion from github.copilot version 1.372.0 or later

ahcorde commented on examples_tf2_py/doc/conf.py:
```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
```
```suggestion
# Copyright 2025 Open Source Robotics Foundation, Inc.
```

ahcorde commented on test_tf2/doc/conf.py:
```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
```
```suggestion
# Copyright 2025 Open Source Robotics Foundation, Inc.
```

ahcorde commented on tf2_bullet/doc/conf.py:
```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
```
```suggestion
# Copyright 2025 Open Source Robotics Foundation, Inc.
```

ahcorde commented on tf2_eigen/doc/conf.py:
```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
```
```suggestion
# Copyright 2025 Open Source Robotics Foundation, Inc.
```

ahcorde commented on tf2_eigen_kdl/doc/conf.py:
```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
```
```suggestion
# Copyright 2025 Open Source Robotics Foundation, Inc.
```

ahcorde commented on tf2_geometry_msgs/doc/conf.py:
```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
```
```suggestion
# Copyright 2025 Open Source Robotics Foundation, Inc.
```

ahcorde commented on tf2_msgs/doc/conf.py:
```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
```
```suggestion
# Copyright 2025 Open Source Robotics Foundation, Inc.
```

ahcorde commented on tf2_py/doc/conf.py:
```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
```
```suggestion
# Copyright 2025 Open Source Robotics Foundation, Inc.
```

ahcorde commented on tf2_ros_py/doc/source/conf.py:
```diff
@@ -1,5 +1,4 @@
-
-# Copyright 2023 Open Source Robotics Foundation, Inc.
+# Copyright 2024 Open Source Robotics Foundation, Inc.
```
```suggestion
# Copyright 2023 Open Source Robotics Foundation, Inc.
```

ahcorde commented on tf2_sensor_msgs/doc/conf.py:
```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
```
```suggestion
# Copyright 2025 Open Source Robotics Foundation, Inc.
```

ahcorde commented on tf2_tools/doc/conf.py:
```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
```
```suggestion
# Copyright 2025 Open Source Robotics Foundation, Inc.
```

rkent committed:
Update copyright date to 2025 in conf.py

Signed-off-by: R Kent James <kent@caspia.com>
Generated-by: Portions of this commit may include code completion from github.copilot version 1.372.0 or later

rkent commented:
I replaced all of the copyright dates to 2025 in a single commit.

ahcorde commented:
Pulls: ros2/geometry2#865
Gist: https://gist.githubusercontent.com/ahcorde/252ae228aa176c4e9c72980756015a32/raw/9d04f37fb52c49dae6d10eefc8b49f9aa2fb989c/ros2.repos
BUILD args:  --packages-above-and-dependencies examples_tf2_py test_tf2 tf2 tf2_bullet tf2_eigen tf2_eigen_kdl tf2_geometry_msgs tf2_kdl tf2_py tf2_ros tf2_ros_py tf2_sensor_msgs tf2_tools
TEST args:  --packages-above examples_tf2_py test_tf2 tf2 tf2_bullet tf2_eigen tf2_eigen_kdl tf2_geometry_msgs tf2_kdl tf2_py tf2_ros tf2_ros_py tf2_sensor_msgs tf2_tools
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/17659
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=26362)](http://ci.ros2.org/job/ci_linux/26362/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=20010)](http://ci.ros2.org/job/ci_linux-aarch64/20010/)
* Linux-rhel [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-rhel&build=6061)](http://ci.ros2.org/job/ci_linux-rhel/6061/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=26102)](http://ci.ros2.org/job/ci_windows/26102/)

ahcorde reviewed (changes requested):
There are some linters failing https://build.ros2.org/job/Rpr__geometry2__ubuntu_noble_amd64/350/

rkent committed:
Fix flake errors in conf.py; delete inadvertently added file from previous commit

Signed-off-by: R Kent James <kent@caspia.com>
Generated-by: Portions of this commit may include code completion from github.copilot version 1.372.0 or later

rkent commented:
OK I fixed the flake errors. Not sure why you saw them now but not previously, as the code in question was copied from an existing conf.py. I also deleted a file I'd added inadvertently in a previous PR.

The auto build with this PR fails, not sure why that is, the error makes no sense with my changes.

ahcorde commented:
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=26382)](http://ci.ros2.org/job/ci_linux/26382/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=20031)](http://ci.ros2.org/job/ci_linux-aarch64/20031/)
* Linux-rhel [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-rhel&build=6082)](http://ci.ros2.org/job/ci_linux-rhel/6082/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=26119)](http://ci.ros2.org/job/ci_windows/26119/)
